core/src/scene/scene.h

```diff
@@ -140,6 +141,8 @@ class Scene {
     std::atomic_ushort pendingTextures{0};
     std::atomic_ushort pendingFonts{0};
 
+    std::vector<SceneError> errors;
```

Comment:
It seems like we only put a maximum of 1 in the list of errors. Maybe instead of a list it can just be a single error state with a default value of "OK".